node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/IP.java

```diff
@@ -140,6 +146,18 @@ public String toString() {
             return nodes;
         }
 
+        /** Returns whether IP address of existing node can be assigned to node */
+        private static boolean canAssignIpOf(Node existingNode, Node node) {
+            if (node.parentHostname().isEmpty()) return false; // Not a child node
```

Comment:
This only works if you add/patch nodes with host first. Maybe
```suggestion
            if (node.parentHostname().isPresent() == existingNode.parentHostname().isPresent()) return false; // Not a parent-child node
            if (node.parentHostname().isEmpty()) return canAssignIpOf(node, existingNode);
```
?

Reply:
Sure, symmetry is great.